rs/ledger_suite/tests/sm-tests/src/lib.rs

```diff
@@ -2643,43 +2668,13 @@ pub fn test_upgrade_serialization(
                     test_upgrade(ledger_wasm_nextmigrationversionmemorymanager.clone());
                     // Test upgrade to memory manager again
                     test_upgrade(ledger_wasm_nextmigrationversionmemorymanager.clone());
-
-                    // Current mainnet ICP and ICRC wasms (V1) can both deserialize from memory manager
-                    match env.upgrade_canister(
-                        ledger_id,
-                        ledger_wasm_mainnet.clone(),
-                        upgrade_args.clone(),
-                    ) {
-                        Ok(_) => {
-                            if !downgrade_to_mainnet_should_succeed {
-                                panic!("Downgrade from memory manager directly to mainnet should fail!")
-                            } else {
-                                // In case this succeeded, we need to upgrade the ledger back to
-                                // the next version (via the current version), so that the
-                                // subsequent upgrade is from
-                                // `ledger_wasm_nextmigrationversionmemorymanager` to
-                                // `ledger_wasm_current`, rather than from `ledger_wasm_mainnet` to
-                                // `ledger_wasm_current` (currently, from V2 -> V1, rather than from
-                                // V0 -> V1).
-                                test_upgrade(ledger_wasm_current.clone());
-                                test_upgrade(ledger_wasm_nextmigrationversionmemorymanager);
-                            }
-                        }
-                        Err(e) => {
-                            if downgrade_to_mainnet_should_succeed {
-                               panic!("Downgrade from memory manager to mainnet should succeed (since mainnet is V1), but failed with error: {}", e)
-                            }
-                            assert!(
-                                e.description().contains("failed to decode ledger state")
-                                    || e.description().contains("Decoding stable memory failed")
-                            )
-                        }
-                    };
                 }
                 // Test deserializing from memory manager
                 test_upgrade(ledger_wasm_current.clone());
-                // Test downgrade to mainnet wasm
-                test_upgrade(ledger_wasm_mainnet.clone());
+                if !migration_to_stable_structures {
```

Comment:
Should we maybe test that the upgrade fails in the else branch?

Reply:
Done!